[BUG] React Fast Refresh warnings in AuthContext and SetupContext

## Bug Description

The Manager UI shows React Fast Refresh warnings during development, preventing full hot module replacement (HMR) optimization for context hooks.

## To Reproduce

1. Run the Manager UI in development mode: `cd manager && pnpm dev`
2. Run the linter: `pnpm lint`
3. Observe the following warnings:

```
src/contexts/AuthContext.tsx
  134:17  warning  Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components  react-refresh/only-export-components

src/contexts/SetupContext.tsx
  67:17  warning  Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components  react-refresh/only-export-components
```

## Expected Behavior

Context files should export only components and context providers to enable full Fast Refresh support during development.

## Actual Behavior

Both `AuthContext.tsx` and `SetupContext.tsx` export custom hooks (`useAuth` and `useSetup`) in the same file as the context provider, which triggers Fast Refresh warnings.

## Screenshots

N/A (console warnings only)

## Environment

- **OS**: macOS 15.1.0 (Darwin 25.1.0)
- **Code Together Version**: Development version
- **AI Tool**: N/A (Manager UI issue)
- **Browser**: N/A (development build)

## Logs

```bash
cd manager && pnpm lint

> code-together-ui@0.1.0 lint
> eslint .

src/contexts/AuthContext.tsx
  134:17  warning  Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components  react-refresh/only-export-components

src/contexts/SetupContext.tsx
  67:17  warning  Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components  react-refresh/only-export-components

✖ 2 problems (0 errors, 2 warnings)
```

## Additional Context

### Proposed Solution

Split the context files to separate concerns:

**Current structure:**
```
src/contexts/
├── AuthContext.tsx        (Provider + useAuth hook)
└── SetupContext.tsx       (Provider + useSetup hook)
```

**Recommended structure:**
```
src/contexts/
├── AuthContext.tsx        (Provider only)
├── AuthContext.hooks.ts   (useAuth hook)
├── SetupContext.tsx       (Provider only)
└── SetupContext.hooks.ts  (useSetup hook)
```

This pattern is recommended by React Fast Refresh documentation to ensure full hot module replacement support during development.

### Impact

- **Development**: Minor - Fast Refresh still works, just with warnings
- **Production**: No impact - build succeeds, warnings are development-only
- **Priority**: Low - Quality of life improvement for developers

## Related Issues

- N/A

## Possible Solution

See "Proposed Solution" above. The fix involves:
1. Creating `AuthContext.hooks.ts` with `useAuth` hook
2. Creating `SetupContext.hooks.ts` with `useSetup` hook
3. Updating imports across the codebase

--- a/manager/src/App.tsx
+++ b/manager/src/App.tsx
@@ -1,6 +1,6 @@
 import { Routes, Route, Navigate } from 'react-router-dom'
-import { useAuth } from './contexts/AuthContext'
-import { useSetup } from './contexts/SetupContext'
+import { useAuth } from './contexts/AuthContext.hooks'
+import { useSetup } from './contexts/SetupContext.hooks'
 import ProtectedRoute from './components/ProtectedRoute'
 import AppLayout from './components/layout/AppLayout'
 import LoginPage from './pages/LoginPage'

--- a/manager/src/components/ProtectedRoute.tsx
+++ b/manager/src/components/ProtectedRoute.tsx
@@ -1,6 +1,6 @@
 import { Navigate, useLocation } from 'react-router-dom'
 import { Box, CircularProgress } from '@mui/material'
-import { useAuth } from '../contexts/AuthContext'
+import { useAuth } from '../contexts/AuthContext.hooks'
 
 interface ProtectedRouteProps {
   children: React.ReactNode

--- a/manager/src/components/layout/Header.tsx
+++ b/manager/src/components/layout/Header.tsx
@@ -15,7 +15,7 @@ import MenuIcon from '@mui/icons-material/Menu'
 import AccountCircleIcon from '@mui/icons-material/AccountCircle'
 import LogoutIcon from '@mui/icons-material/Logout'
 import CodeIcon from '@mui/icons-material/Code'
-import { useAuth } from '../../contexts/AuthContext'
+import { useAuth } from '../../contexts/AuthContext.hooks'
 
 interface HeaderProps {
   onMenuClick: () => void

--- a/manager/src/components/layout/Sidebar.tsx
+++ b/manager/src/components/layout/Sidebar.tsx
@@ -13,7 +13,7 @@ import BarChartIcon from '@mui/icons-material/BarChart'
 import HistoryIcon from '@mui/icons-material/History'
 import GroupIcon from '@mui/icons-material/Group'
 import StorageIcon from '@mui/icons-material/Storage'
-import { useAuth } from '../../contexts/AuthContext'
+import { useAuth } from '../../contexts/AuthContext.hooks'
 
 interface NavItem {
   label: string

--- a/manager/src/contexts/AuthContext.hooks.ts
+++ b/manager/src/contexts/AuthContext.hooks.ts
@@ -1,0 +1,10 @@
+import { useContext } from 'react'
+import { AuthContext } from './AuthContext'
+
+export function useAuth() {
+  const context = useContext(AuthContext)
+  if (context === undefined) {
+    throw new Error('useAuth must be used within an AuthProvider')
+  }
+  return context
+}

--- a/manager/src/contexts/AuthContext.tsx
+++ b/manager/src/contexts/AuthContext.tsx
@@ -1,6 +1,5 @@
 import {
   createContext,
-  useContext,
   useState,
   useEffect,
   useCallback,
@@ -24,7 +23,8 @@ interface AuthContextValue extends AuthState {
   refreshUser: () => Promise<void>
 }
 
-const AuthContext = createContext<AuthContextValue | undefined>(undefined)
+// eslint-disable-next-line react-refresh/only-export-components
+export const AuthContext = createContext<AuthContextValue | undefined>(undefined)
 
 interface AuthProviderProps {
   children: ReactNode
@@ -129,12 +129,4 @@ export function AuthProvider({ children }: AuthProviderProps) {
       {children}
     </AuthContext.Provider>
   )
-}
-
-export function useAuth(): AuthContextValue {
-  const context = useContext(AuthContext)
-  if (context === undefined) {
-    throw new Error('useAuth must be used within an AuthProvider')
-  }
-  return context
 }

--- a/manager/src/contexts/SetupContext.hooks.ts
+++ b/manager/src/contexts/SetupContext.hooks.ts
@@ -1,0 +1,10 @@
+import { useContext } from 'react'
+import { SetupContext } from './SetupContext'
+
+export function useSetup() {
+  const context = useContext(SetupContext)
+  if (context === undefined) {
+    throw new Error('useSetup must be used within a SetupProvider')
+  }
+  return context
+}

--- a/manager/src/contexts/SetupContext.tsx
+++ b/manager/src/contexts/SetupContext.tsx
@@ -1,6 +1,5 @@
 import {
   createContext,
-  useContext,
   useState,
   useEffect,
   type ReactNode,
@@ -17,7 +16,8 @@ interface SetupContextValue extends SetupState {
   checkSetupStatus: () => Promise<void>
 }
 
-const SetupContext = createContext<SetupContextValue | undefined>(undefined)
+// eslint-disable-next-line react-refresh/only-export-components
+export const SetupContext = createContext<SetupContextValue | undefined>(undefined)
 
 interface SetupProviderProps {
   children: ReactNode
@@ -62,12 +62,4 @@ export function SetupProvider({ children }: SetupProviderProps) {
       {children}
     </SetupContext.Provider>
   )
-}
-
-export function useSetup(): SetupContextValue {
-  const context = useContext(SetupContext)
-  if (context === undefined) {
-    throw new Error('useSetup must be used within a SetupProvider')
-  }
-  return context
 }

--- a/manager/src/pages/DashboardPage.tsx
+++ b/manager/src/pages/DashboardPage.tsx
@@ -14,7 +14,7 @@ import {
 import Grid from '@mui/material/Grid2'
 import RefreshIcon from '@mui/icons-material/Refresh'
 import { dashboardApi } from '../api/dashboard'
-import { useAuth } from '../contexts/AuthContext'
+import { useAuth } from '../contexts/AuthContext.hooks'
 import ActivityChart from '../components/charts/ActivityChart'
 import ProviderRankingChart from '../components/charts/ProviderRankingChart'
 import MemberStatsTable from '../components/MemberStatsTable'

--- a/manager/src/pages/LoginPage.tsx
+++ b/manager/src/pages/LoginPage.tsx
@@ -10,7 +10,7 @@ import {
   Alert,
   CircularProgress,
 } from '@mui/material'
-import { useAuth } from '../contexts/AuthContext'
+import { useAuth } from '../contexts/AuthContext.hooks'
 
 export default function LoginPage() {
   const navigate = useNavigate()

--- a/manager/src/pages/ProfilePage.tsx
+++ b/manager/src/pages/ProfilePage.tsx
@@ -9,7 +9,7 @@ import {
   Divider,
 } from '@mui/material'
 import Grid from '@mui/material/Grid2'
-import { useAuth } from '../contexts/AuthContext'
+import { useAuth } from '../contexts/AuthContext.hooks'
 
 export default function ProfilePage() {
   const { user } = useAuth()

--- a/manager/src/pages/SetupPage.tsx
+++ b/manager/src/pages/SetupPage.tsx
@@ -18,7 +18,7 @@ import {
 import { Visibility, VisibilityOff, ContentCopy } from '@mui/icons-material'
 import { setupApi } from '../api/setup'
 import { getErrorMessage } from '../api/client'
-import { useSetup } from '../contexts/SetupContext'
+import { useSetup } from '../contexts/SetupContext.hooks'
 
 const steps = ['Organization', 'Administrator Account', 'Complete']
 

